stacks-signer/src/main.rs

```diff
@@ -79,22 +85,66 @@ fn write_chunk_to_stdout(chunk_opt: Option<Vec<u8>>) {
     }
 }
 
-fn spawn_running_signer(
-    path: &PathBuf,
-    command: RunLoopCommand,
-    cmd_recv: Receiver<RunLoopCommand>,
-) -> RunningSigner<StackerDBEventReceiver, Vec<Point>> {
+// Spawn a running signer and return its handle, command sender, and result receiver
+fn spawn_running_signer(path: &PathBuf) -> SpawnedSigner {
     let config = Config::try_from(path).unwrap();
+    let (cmd_send, cmd_recv) = channel();
+    let (res_send, res_recv) = channel();
     let ev = StackerDBEventReceiver::new(vec![config.stackerdb_contract_id.clone()]);
-    let runloop: RunLoop<FrostCoordinator> = RunLoop::new(&config, command);
+    let runloop: RunLoop<FrostCoordinator> = RunLoop::from(&config);
     let mut signer: Signer<
         RunLoopCommand,
-        Vec<Point>,
+        Vec<OperationResult>,
         RunLoop<FrostCoordinator>,
         StackerDBEventReceiver,
-    > = Signer::new(runloop, ev, cmd_recv);
+    > = Signer::new(runloop, ev, cmd_recv, res_send);
     let endpoint = config.node_host;
-    signer.spawn(endpoint).unwrap()
+    let running_signer = signer.spawn(endpoint).unwrap();
+    SpawnedSigner {
+        running_signer,
+        cmd_send,
+        res_recv,
+    }
+}
+
+// Process a DKG result
+fn process_dkg_result(dkg_res: &[OperationResult]) {
+    if dkg_res.len() != 1 {
```

Comment:
Why doesn't this method just take a single `&OperationResult` instead of a slice reference?

Reply:
Same question for `process_sign_result()` below

Reply:
Because the run one pass can in theory return multiple items (it doesn't right now because of how we do things sequentially).  I added this check because we would not expect more than one DKG result or more than one sign. However, handling it internally to this function call made more sense to me/removed the most amount of duplicate code. In theory of course I can assert the size of the list outside of this function call and access the first element in the list, but then we will have more duplicate code in handle_dkg, handle_sign, and handle_dkg_sign.

Reply:
I think it's fine as long as it's just part of `main`'s CLI handling.  I was mainly curious as to why the runloop itself doesn't just send a single `OperationResult` back instead of `Vec<OperationResult>`, because it seemed to be the case that the code always expected a `Vec` with exactly one element.  But if in the future there will be multiple items, then this is fine.